### PR TITLE
Add API to configure CircuitOptions

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Components/Server/src/DependencyInjection/ServerSizeBlazorBuilderExtensions.cs [hunk 1]
```diff
@@ -12,6 +12,29 @@ namespace Microsoft.Extensions.DependencyInjection
     /// </summary>
     public static class ServerSizeBlazorBuilderExtensions
     {
+        /// <summary>
+        /// Adds options to configure circuits.
+        /// </summary>
+        /// <param name="builder">The <see cref="IServerSideBlazorBuilder"/>.</param>
+        /// <param name="configure">A callback to configure <see cref="CircuitOptions"/>.</param>
+        /// <returns>The <see cref="IServerSideBlazorBuilder"/>.</returns>
+        public static IServerSideBlazorBuilder AddCircuitOptions(this IServerSideBlazorBuilder builder, Action<CircuitOptions> configure)
```
Comment (pranavkm) [inline src/Components/Server/src/DependencyInjection/ServerSizeBlazorBuilderExtensions.cs]: Makes it a bit more discoverable. This is fairly undiscoverable now. How do we feel about `AddServerSideBlazor` optionally accepting a callback to configure `CircuitOptions`?

File: src/Components/Server/ref/Microsoft.AspNetCore.Components.Server.netcoreapp3.0.cs [hunk 2]
```diff
@@ -103,6 +103,7 @@ public partial interface IServerSideBlazorBuilder
     }
     public static partial class ServerSizeBlazorBuilderExtensions
```
Comment (SteveSandersonMS) [inline src/Components/Server/ref/Microsoft.AspNetCore.Components.Server.netcoreapp3.0.cs]: I know it predates your PR, but there's a typo in the class name. Server "size"?

Comment (SteveSandersonMS) [reply src/Components/Server/ref/Microsoft.AspNetCore.Components.Server.netcoreapp3.0.cs]: The typo is also in the filename. Would you mind fixing it as part of this PR?

**Review by SteveSandersonMS — APPROVED** [review]
Looks like a good improvement to me.

Without this PR, is the only way to configure `CircuitOptions` calling `services.Configure<CircuitOptions>(...)` manually? If so I agree that's extremely hard to discover.

File: src/Components/Server/ref/Microsoft.AspNetCore.Components.Server.netcoreapp3.0.cs (hunk 2 above)
Comment (javiercn) [reply src/Components/Server/ref/Microsoft.AspNetCore.Components.Server.netcoreapp3.0.cs]: lol. We should change the product to be ServerSizeBlazor 😄

File: src/Components/Server/src/DependencyInjection/ServerSizeBlazorBuilderExtensions.cs (hunk 1 above)
Comment (javiercn) [reply src/Components/Server/src/DependencyInjection/ServerSizeBlazorBuilderExtensions.cs]: I think that would be good. Its a bit jarring to see AddServer**Size**Blazor and the options type name to be different, but I can live with that.

**Review by javiercn — APPROVED** [review]

**Comment by pranavkm** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=223774&view=ms.vss-test-web.build-test-results-tab&runId=5514352&resultId=112852&paneView=debug

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2667

**Pull request merged**